embedded-workshop-book/src/bsc-exercise.md

```diff
@@ -1,37 +1,69 @@
 # BSC Exercise
 
-In this exercise you will learn how to write a board support crate. 
-The bsc template will already contain the led and timer implementation. 
-The radio and USB/Power implementations will be deleted, because that just takes up unnecessary space and adds to confusion. 
+In this exercise you will learn how to write a board support crate by implementing buttons and the UARTE peripheral. 
 
+The template `down-the-stack/dk_bsc/src/lib.rs` already contains the LED and Timer implementations. Add your code to the designated lines. You'll find a //todo! there. 
 
-## Learning goals
-* implement buttons functionality 
-* uarte implementation
-* impl blocks, associated functions, methods
-* generate docs!
+You can test after each step by running the following command out of `down-the-stack/apps`
+```
+cargo run --bin hello
+```
+This program will not call any of the functions you are implementing, so it does not matter if they are incomplete. It will refuse to build if there are errors present in the `lib.rs`!
 
+`down-the-stack/dk_bsc/src/lib_solution.rs` contains the full solution code. 
 
-## Steps
-### Write a button implementation
-* add field in the board struct
-* add struct for all buttons
-* add struct for the single button
+Note for trainer: Introduction to the exercise is a guided tour through the template, and it's architecture. Make the participants aware of the placeholders for their implementations. run the hello example on the unmodified lib. 
```

Comment:
Nits:
Introction... -> "The introduction to the exercise is a guided tour through the template, and its architecture."
implementations. run -> "Implementations. Run the ..."